axelrod/mock_player.py

```diff
@@ -8,16 +8,26 @@
 
 
 class MockPlayer(Player):
-    """Creates a mock player that plays a given sequence of actions. If
-    no actions are given, plays like Cooperator. Used for testing.
+    """Creates a mock player that cycles through a given 
+    sequence of actions. If no actions are given, 
+    plays like Cooperator. Used for testing.
+
+    Parameters
+    ----------
+    actions: List[Action], default []
+        The sequence of actions played by the mock player.
+    attributes: dict, default {}
+        A dictionary of player attributes.
     """
 
     name = "Mock Player"
 
-    def __init__(self, actions: List[Action] = None) -> None:
+    def __init__(self, actions: List[Action] = None, classifier: dict = None) -> None:
         super().__init__()
         if not actions:
             actions = []
+        if classifier:
```

Comment:
Should this be attributes?

Reply:
Also, not sure you need the if here?

Reply:
turns out the `if` is load bearing - the `super().__init__()` sets classifier to `{}` so without the if statement it already takes a default there (as opposed to a usual construct like defaulting it to None and saying `if classifier is None; classifier = {}`). reverted

Reply:
I still don't quite understand this one. `classifier` is local so how can `Player.__init__` set it? I could see `self.classifier` being set by `Player.__init__`, and if that's the case, maybe we should be passing `classifier` in as a kwarg to `Player`?

Reply:
I checked because I wasn't sure either, but it turns out the local variable `classifier` in the body of the class and the attribute `self.classifier` in the instance attributes are *the same thing*! I was surprised to see it works like this. Here's an example:

```python
>>> class ExampleClass:
... 
...     attribute1 = "Hello, World!"
...
>>> example = ExampleClass()
>>> print(example.attribute1)
"Hello, World!"
```
The main difference, of course, is that local variables can be set externally through instantiation parameters in `__init__` but can only be set in the body of the class when the class source code is written. I'd argue this is a good thing (I don't think classifiers should be exposed to the user). As far as I can tell, all `instance.attribute` does is get the local variable `attribute`from the scope of `instance`.

Reply:
actually, looking further into these docs, [the attribute in the body is shared by all instances](https://docs.python.org/3/tutorial/classes.html#class-and-instance-variables); so turns out that not only is the if statement necessary it's actually insufficient (e.g. we need the `if Classifier is None` etc. etc.)